wolfcrypt/src/cmac.c

```diff
@@ -305,6 +305,58 @@ int wc_CmacFinal(Cmac* cmac, byte* out, word32* outSz) {
     return ret;
 }
 
+
+int wc_AesCmacGenerate_ex(Cmac* cmac,
+                          byte* out, word32* outSz,
+                          const byte* in, word32 inSz,
+                          const byte* key, word32 keySz,
+                          int type, void* heap, int devId)
```

Comment:
Just thought of this.  The function is AEScmacgenerate, so why do we accept the "type" parameter?  Maybe we should drop it and simply force it to WC_CMAC_AES in the init call below.  Same with the verify.  Otherwise, we really should check that it is the value we expect OR we expose generic version of cmacgenerate that is invoked by AEScmacgenerate.

Reply:
I just followed the signature of the other _ex invocations, but I agree about the hard-coded AES dependencies. Removed.